### PR TITLE
cmake: move ti_lm3s6965-specific code into the soc directory

### PR DESCRIPTION
**Pull request by SebastianBoe**
As far as possible SoC-specific code should be contained to it's
directory and core systems should not be referencing specific
SoCs. This keeps the common code clean and makes it easier to maintain
out-of-tree SoCs.

To this end we move CONFIG_SOC_TI_LM3S6965_QEMU's soft-reset related
code out of arch/arm/core/cortex_m and into arch/arm/soc/ti_lm3s6965.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Review by nashif — APPROVED** [review]

File: soc/arm/atmel_sam/same70/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,6 @@
 zephyr_sources(
   soc.c
   soc_config.c
+  reboot.S
+  sys_arch_reboot.c
```
Comment (MaureenHelm) [inline soc/arm/atmel_sam/same70/CMakeLists.txt]: Why are you modifying this for Atmel? Shouldn't it be in `soc/arm/ti_lm3s6965/CMakeLists.txt`?

**Review by MaureenHelm — DISMISSED** [review]

File: soc/arm/atmel_sam/same70/CMakeLists.txt (hunk 1 above)
Comment (SebastianBoe) [reply soc/arm/atmel_sam/same70/CMakeLists.txt]: This was a mistake. I'm not sure how my testing didn't catch this ...

Comment (SebastianBoe) [reply soc/arm/atmel_sam/same70/CMakeLists.txt]: Strange, locally this looks like

arch/arm/soc/ti_lm3s6965/CMakeLists.txt    |  2 +

There must be something strange going on with Github.

Comment (nashif) [reply soc/arm/atmel_sam/same70/CMakeLists.txt]: my mistake, rebase was too optimistic, just updated.

Comment (nashif) [reply soc/arm/atmel_sam/same70/CMakeLists.txt]: > There must be something strange going on with Github.

no, it is me, trying to resolve all of those conflicts after the soc move PR  :)

I did a rebase and git decided to make the change with the atmel board for whatever reason.

Comment (SebastianBoe) [reply soc/arm/atmel_sam/same70/CMakeLists.txt]: In any case. It is fixed now. Good catch @MaureenHelm .

Comment (SebastianBoe) [reply soc/arm/atmel_sam/same70/CMakeLists.txt]: Ah, no, now I made a mistake ...

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**